### ISSUE TITLE
Last node in color map is always black

### ISSUE THREAD

**Issue by moreApi** [scenerygraphics/scenery#750]
as seen for example here:
![java_t13jR7HcXm](https://private-user-images.githubusercontent.com/83579186/328846765-05e42f2b-e950-4a5e-8239-11d2d7d2f91b.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUxNzk1MjgsIm5iZiI6MTcxNTE3OTIyOCwicGF0aCI6Ii84MzU3OTE4Ni8zMjg4NDY3NjUtMDVlNDJmMmItZTk1MC00YTVlLTgyMzktMTFkMmQ3ZDJmOTFiLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA4VDE0NDAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU1OTMzNjhmYzJjNmRmNWQ3ZjM3MDFmOTI2ZGI0N2RiYjY1MTJlNzJhNTA2OTM1ZDQyMzExYjhhMWYxZjEzMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BjxdZ1UBtC5dqxDIJVhKs0HBLjWFVYl20U37fFhKJmA)

**Comment by odinsbane** [comment]
@moreApi Your example is missing. I seem to be having an issue where textures turn black. If I select a new colormap it will briefly show color, then turn black again. Maybe it is related?

**Comment by smlpt** [comment]
@odinsbane the volume turning black is likely caused by [this bug](https://github.com/scenerygraphics/scenery/issues/757), for me it happens as soon as I hover over the color map editor.

**Comment by odinsbane** [comment]
Exactly. That is the issue I am having. I don't know if it is related to this issue.

**Comment by odinsbane** [comment]
This error is caused by [Colormap#sample](https://github.com/scenerygraphics/scenery/blob/78ad6b81655ff17fde95a1b1f0d600e28e9a7798/src/main/kotlin/graphics/scenery/volumes/Colormap.kt#L34)

    val bufferPosition: Float = position.coerceIn(0.0f, 1.0f) * width

Indexes are 0 based so they should go from 0 to width - 1. This is 0 to width, but since the Colormap can have multiple bands it wraps back to the next band... which is black again.
